packages/config-loader/src/lib/secrets.ts

```diff
@@ -111,24 +104,30 @@ export async function readSecret(
     return ctx.env[secret.env];
   }
   if ('data' in secret) {
-    const ext = extname(secret.data);
+    const url =
+      'path' in secret ? `${secret.data}#${secret.path}` : secret.data;
+    const [filePath, dataPath] = url.split(/#(.*)/);
+    if (!dataPath) {
+      throw new Error(
```

Comment:
So if you actually want the entire file contents you need to specify "."? Does this work for a file that contains only a single raw string?

Reply:
Oh yeah prolly makes sense to allow the path to not be defined and just return the entire parsed file then. The file will be parsed according to the extension, so I guess a single raw string might work in yaml, but not json. Should use `$file` for that anyway though

Reply:
yeah you may want to separate out file, yaml and json like the other one for this reason right

Reply:
Why is that?

Reply:
Oh actually gonna skip this for now. Realized that secrets are expected to always be strings and nothing else right now, so this would be a later change.

Reply:
Hm, i'm being unclear. I'm just saying that the placeholder processor has text and json/yaml (whether they are a merged $data or two separate) as two separate concepts for this reason.

Reply:
I know it's a little different in this code here, since for the placeholder processor, the placeholders describe the nature of the DATA, since the source is always the same (some url). Here, the placeholders describe also the nature of the SOURCE since there are at least two different possibilities (env and file).

Reply:
Still not following, you could just as well use `$file` and `$data` the way they work here for the placeholder processors too? With the exception that `$data` would actually be able to return objects etc.

Reply:
Mmwell maybe I misread the code then. Anyway, not related to the PR. :) Some other time